src/main/java/org/openstreetmap/atlas/geography/atlas/raw/creation/OsmPbfCounter.java

```diff
@@ -106,12 +193,243 @@ else if (rawEntity instanceof Bound)
+     * any ways that meet the above criteria. The while loop terminates if we've exceeded the
+     * maximum number of extensions or if we haven't added anything during the previous iteration.
+     */
+    private void bringInConnectedOutsideWays()
+    {
+        if (this.loadingOption.isLoadWaysSpanningCountryBoundaries())
+        {
+            int extensionCounter = 0;
+            final AtomicBoolean addedNewEdge = new AtomicBoolean(true);
+            while (extensionCounter < MAXIMUM_NETWORK_EXTENSION && addedNewEdge.get())
+            {
+                logger.trace("Adding connected ways outside boundary pass {}", extensionCounter);
+                addedNewEdge.set(false);
+                this.waysToExclude.values().stream().filter(this::isHighwayOrFerry).forEach(way ->
+                {
+                    final List<WayNode> wayNodes = way.getWayNodes();
+                    for (final WayNode wayNode : wayNodes)
+                    {
+                        final long identifier = wayNode.getNodeId();
+                        if (this.nodeIdentifiersBroughtInByWaysOrRelations.contains(identifier))
+                        {
+                            final Long startNodeIdentifier = wayNodes.get(0).getNodeId();
+                            final Long endNodeIdentifier = wayNodes.get(wayNodes.size() - 1)
+                                    .getNodeId();
+
+                            // Look at the start point
+                            if (startNodeIdentifier != identifier)
+                            {
+                                // The start node is a new node outside of the network
+                                this.nodeIdentifiersBroughtInByWaysOrRelations
+                                        .add(startNodeIdentifier);
+                            }
+
+                            // Look at the end point
+                            if (endNodeIdentifier != identifier)
+                            {
+                                // The end node is a new node outside of the network
+                                this.nodeIdentifiersBroughtInByWaysOrRelations
+                                        .add(endNodeIdentifier);
+                            }
+
+                            // Add way and its members
+                            logger.trace("Adding connected way with identifier {}", way.getId());
+                            this.wayIdentifiersToInclude.add(way.getId());
+                            addWayNodes(this.nodeIdentifiersBroughtInByWaysOrRelations, way);
+                            addedNewEdge.set(true);
+                            break;
+                        }
+                    }
+                });
+                extensionCounter++;
+            }
+        }
+    }
+
+    private boolean isHighwayOrFerry(final Way way)
+    {
+        final Taggable taggableWay = new Taggable()
```

Comment:
Isn't there a converter for osmosis ways to taggable? I am not sure. Might be dreaming

Reply:
There isn't! But doing some digging, I can leverage TagMap which handles translation between a Collection<Tag> to a Taggable object. Let me know what you think - simplifies the code quite a bit!